runners/spark/src/main/java/org/apache/beam/runners/spark/util/PipelineHelper.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.spark.util;
+
+import org.apache.beam.runners.spark.SparkPipelineOptions;
+
+public class PipelineHelper {
+
+  public static boolean isLocalSpark(SparkPipelineOptions options) {
```

Comment:
Can we better move this method to `SparkCommonPipelineOptions` it does not make sense to create a full class for this, and also it can be reused from there by the structured streaming runner.
Maybe worth to bring also the Structured Streaming fix into this PR so we don't have inconsistencies in the Nexmark runtimes in the CI.

Reply:
Optional: Maybe we can rename the method too, to `isLocalMaster` or something like that, because this is what it is detecting.

Reply:
> Can we better move this method to `SparkCommonPipelineOptions` it does not make sense to create a full class for this, and also it can be reused from there by the structured streaming runner.

 `SparkCommonPipelineOptions`  is an interface, so the `isLocalSpark` method would not be accessible from outside the interface to `SparkRunner` class hence adding a new utility class.
> Maybe worth to bring also the Structured Streaming fix into this PR so we don't have inconsistencies in the Nexmark runtimes in the CI.
Good point, will cherry pick